nixos/tests/docuseal-postgres.nix

```diff
@@ -0,0 +1,47 @@
+import ./make-test-python.nix (
+  { lib, ... }:
+  {
+    name = "docuseal";
+    meta.maintainers = with lib.maintainers; [
+      etu
+      stunkymonkey
+    ];
+
+    nodes.machine =
+      { pkgs, ... }:
+      {
+        services.docuseal = {
+          enable = true;
+          port = 80;
+          secretKeyBaseFile = pkgs.writeText "secret" "23bec595a1658d136d532af1365b40024b662c0862e9cdf14fd22c0afaeb0dd6322b114fa35bd82e564bae44a896b5abef3a66afd61e1382b8ebd579e2c5c17f";
+          extraConfig.DATABASEURL = "postgresql://docuseal:db-secret@127.0.0.1:5432/docuseal";
+        };
+        services.postgresql = {
+          package = pkgs.postgresql;
+          enable = true;
+          ensureDatabases = [ "docuseal" ];
+          ensureUsers = [
+            {
+              name = "docuseal";
+              ensureDBOwnership = true;
+            }
+          ];
+          initialScript = pkgs.writeText "postgresql-password" ''
+            CREATE ROLE docuseal WITH LOGIN PASSWORD 'db-secret' CREATEDB;
+          '';
+        };
+
+        systemd.services."docuseal-config" = {
+          requires = [ "postgresql.service" ];
+          after = [ "postgresql.service" ];
+        };
+      };
+
+    testScript = ''
+      machine.wait_for_unit("multi-user.target")
+      machine.wait_for_open_port(80)
```

Comment:
Please also throw in a wait for the unit `docuseal.service`. The reason is that this will fail loudly and immediately if the unit does not come up e.g. due to a mistake in the configuration, instead of timing out after a long time because port 80 does not get opened.

Reply:
Generally, adding global timeouts will be useful for CI throughput